presto-main/src/main/java/com/facebook/presto/server/HttpRequestSessionFactory.java

```diff
@@ -222,6 +228,18 @@ public Session createSession(
         return sessionProperties;
     }
 
+    private Set<String> parseTags(HttpServletRequest servletRequest)
+    {
+        String tags = servletRequest.getHeader(PRESTO_TAGS);
+        if (tags != null) {
+            Splitter splitter = Splitter.on(',').trimResults().omitEmptyStrings();
```

Comment:
You can simplify this code by doing nullToEmpty() on "tags". Then you don't need an if statement.